doc/reference_type.xml

```diff
@@ -2,43 +2,65 @@
   <sect1 id="PostGIS_Types">
     <sect1info>
     <abstract>
-    <para>This section lists the PostgreSQL data types installed by PostGIS.  Note we describe the casting behavior of these which is very
-	important especially when designing your own functions.
+    <para>This section lists the custom PostgreSQL
+		data types installed by PostGIS to represent spatial data.
     </para>
-    <para>A Cast is when one type is coerced into another type.  PostgreSQL
-	is unique from most databases in that it allows you to define casting behavior for custom types and the functions used for casting.
-	A cast can be specified as automatic in which case, you do not have to do a CAST(myfoo As otherfootype) or myfoo::otherfootype
-	if you are feeding it to a function that only works with otherfootype and there is an automatic cast in place for it.
-    </para>
-    <para>The danger of relying on automatic cast behavior is when you have an overloaded function say one that takes a box2d and one that takes a box3d
-	but no geometry.  What happens is that both functions are equally good to use with geometry since geometry has an autocast for both
-	-- so you end up with an ambiguous function error. To force PostgreSQL to choose, you do a CAST(mygeom As box3d)  or mygeom::box3d.</para>
+    <para>Each data type describes its type casting behaviour.
+		A <ulink url="https://www.postgresql.org/docs/current/sql-expressions.html#SQL-SYNTAX-TYPE-CASTS">type cast</ulink> converts values of one data type into another type.
+		PostgreSQL allows defining casting behavior for custom types, along with the functions used to convert type values.
+		Casts can be defined as <emphasis role="bold">automatic</emphasis>.
+		This allows PostgreSQL to automatically convert a function argument to a type supported by the function.
+		Casts can also be specified <emphasis role="bold">explicitly</emphasis>,
+		using the syntax <varname>CAST(myval As sometype)</varname>  or <varname>myval::sometype</varname>.
+		This allows avoiding an issue that can occur when using an overloaded function which does not support a given type
+		(for example, one that accepts a box2d or a box3d, but not a geometry.)
+		Since geometry has an automatic cast to both box types, this will produce an "ambiguous function" error.
+	  To prevent the error use an explicit cast to the desired box type.</para>
 
-    <para><emphasis>At least as of PostgreSQL 8.3</emphasis> - Everything can be CAST to text (presumably because of the magical unknown type), so no defined CASTS for that need to be present for you to CAST an object to text.</para>
+    <para><emphasis>(At least as of PostgreSQL 8.3)</emphasis> Everything can be CAST to <varname>text</varname>, so CASTs to text do not need to be specified.</para>
```

Comment:
Postgres 8.3 is dead since 2013. 
```suggestion
    <para>Everything can be CAST to <varname>text</varname>, so CASTs to text do not need to be specified.</para>
```